src/main/cpp/lang/xml.cpp

```diff
@@ -0,0 +1,478 @@
+#include "ulight/impl/lang/xml.hpp"
+#include "ulight/impl/lang/html.hpp"
+#include "ulight/impl/lang/xml_chars.hpp"
+
+#include "ulight/impl/buffer.hpp"
+#include "ulight/impl/highlight.hpp"
+#include "ulight/impl/highlighter.hpp"
+
+#include "ulight/impl/unicode_algorithm.hpp"
+#include "ulight/ulight.hpp"
+
+#include <cctype>
+#include <cmath>
+#include <functional>
+#include <iterator>
+#include <string>
+
+namespace ulight {
+
+namespace xml {
+
+namespace {
+
+constexpr std::u8string_view comment_prefix = u8"<!--";
+constexpr std::u8string_view comment_suffix = u8"-->";
+constexpr std::u8string_view illegal_comment_sequence = u8"--";
+constexpr std::u8string_view cdata_section_prefix = u8"<![CDATA[";
+constexpr std::u8string_view cdata_section_suffix = u8"]]>";
+
+[[nodiscard]]
+bool contains_xml_string(std::u8string_view str)
+{
+
+    if (str.length() < 3) {
+        return false;
+    }
+
+    for (std::size_t i = 0; i <= str.length() - 3; i++) {
```

Comment:
For bug prevention reasons, it is better to do `i + 2 < str.length()`. This style tells you that performing `i + 2` in the code below is safe.